build/cloud/ha/README.md

```diff
@@ -0,0 +1,156 @@
+# The HA of CloudCore(deployed in k8s cluster)
+
+## Determine the virtual IP of CloudCore
+
+Determine a VIP that the CloudCore service exposed to the edge nodes. Here we recommend `keepalived` to do that. You had better directly schedule pods to specific number of nodes by `nodeSelector` when using  `keepalived`. And you have  to install `keepalived` in each of nodes where CloudCore runs. The configuration of `keepalived` is shown in the end. Here suppose the VIP is 10.10.102.242.
+
+The use of `nodeSelector` is as follow:
+
+```bash
+kubectl label nodes [nodename] [key]=[value]  # label the nodes where the cloudcore will run
+```
+
+modify the term of `nodeselector`:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cloudcore
+spec:
+  template:
+    spec:
+      nodeSelector: # configure the nodeSelector here!
+        [key]: [value]
+```
+
+## Create k8s resources
+
+The manifests and scripts in `github.com/kubeedge/kubeedge/build/cloud/ha` will be used, so place these files to somewhere you can kubectl with.
+
+First, ensure your k8s cluster can pull cloudcore image. If the image not exist. We can make one, and push to your registry.
+
+```bash
+cd $GOPATH/src/github.com/kubeedge/kubeedge
+make cloudimage
+```
+
+We create k8s resources from the manifests in name order. Before creating, **check the content of each manifest to make sure it meets your environment.**
+
+**Note:** Now the follow manifests don't support `kubectl logs` command yet. If need, you have to make more configuration manually.
+
+### 02-ha-configmap.yaml
+
+Configure the VIP address of CloudCore which is exposed to the edge nodes in the `advertiseAddress`, which will be added to SANs in cert of CloudCore. For example:
+
+```yaml
+modules:
+      cloudHub:
+        advertiseAddress:
+         - 10.10.102.242
+```
+
+**Note:** If you want to reset the CloudCore, run this before creating k8s resources:
+
+```bash
+kubectl delete namespace kubeedge
+```
+
+Then create k8s resources:
+
+```shell
+cd build/cloud/ha
+for resource in $(ls *.yaml); do kubectl create -f $resource; done
+```
+
+## keepalived
```

Comment:
keepalived is just one of the options for setting up loadbalancers for cloudcore instances.

Reply:
> keepalived is just one of the options for setting up loadbalancers for cloudcore instances.



> keepalived is just one of the options for setting up loadbalancers for cloudcore instances.

yep, I will mark it.